[DPE-4848;DPE-3552] Update documentation for GPU support and custom CA (#33)
canonical/spark-k8s-bundle: releases/3.4/yaml/docs/how-to/h-spark-gpu.md

First, we need to create a pod template to limit the amount of gpu per container.

Edit the pod manifest file (we'll refer to it as ```gpu_executor_template.yaml```) by adding the following content:

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
    - name: executor
      resources:
        limits:
          nvidia.com/gpu: 1
```

### Submitting  a Spark job with gpu acceleration

With the usage of the `spark-client` snap we can now submit the desired Spark job. 
In order to run the job with gpu acceleration, some configuration options need to be used:

```shell
spark.executor.resource.gpu.amount=1
spark.task.resource.gpu.amount=1
spark.rapids.memory.pinnedPool.size=1G
spark.plugins=com.nvidia.spark.SQLPlugin
spark.executor.resource.gpu.discoveryScript=/opt/getGpusResources.sh
spark.executor.resource.gpu.vendor=nvidia.com
spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark-gpu:3.4-22.04_edge
spark.kubernetes.executor.podTemplateFile=gpu_executor_template.yaml
```

> The Spark configuration options can be set at the spark service account with the spark-client snap to use them on all the jobs. Please have a look at this [guide](https://discourse.charmhub.io/t/spark-client-snap-how-to-manage-spark-accounts/8959) on how to manage options at service account level. To have more information on how the `spark-client` manages configuration options please have a look [here](https://discourse.charmhub.io/t/spark-client-snap-explanation-hierarchical-configuration-handling/8956). 

Reviewer: This is a practical How to guide. Can we directly provide instructions on how to set the required options?

Reviewer 2: The practical guide is the one that is linked. I feel that it would be repetition  to also put the information here, beside the user has different way to set these configuration (via python, snap or configuration hub)

Reviewer: In this case, I'd put a single example of how to do the step in one way (any of them, but ideally, the easiest one), and the link to find more details and alternatives. Otherwise, it feels like we are describing what needs to be done without providing step-by-step instructions on how to do that. It's not terrible, but less than ideal.

Reviewer 2: Ironically, it used to be this way, but I asked @welpaolo to make it more generic, since I personally would have let the how-to to be rather general, and not nudge people to use one method over another. 

But, hey, you are the documentation author, so this should be your call. I have reverted back the change to something similar to @welpaolo original version.

> The aforementioned options are the minimal set of configurations that are needed to enable the Spark Rapids plugin. The full list of available options are available [here](https://nvidia.github.io/spark-rapids/docs/configs.html).

Reviewer: ```suggestion
The options above are the minimal set that is needed to enable the Spark Rapids plugin. 
For more information on available options, see the [full list](https://nvidia.github.io/spark-rapids/docs/configs.html).
```
Some suggestions on how to improve the wording here.
We should avoid using words like `here` as links.
